File: python/leetcode/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures):
        """
        :type temperatures: List[int]
        :rtype: List[int]
        """
        result = []
        for i in range(len(temperatures)):
            ti = i
            c = 0
            greater_found = False
            o_temp = temperatures[i]
            while ti < len(temperatures):
                if temperatures[ti] > o_temp:
                    greater_found = True
                    break
                else:
                    c += 1
                ti += 1
            result.append(c if greater_found else 0)
        return result

File: java/leetcode/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] result = new int[temperatures.length];
        int ti = 0;
        int oTemp = 0;
        int c = 0;
        boolean greaterFound = false;
        for (int i = 0; i < temperatures.length; i++) {
            ti = i;
            oTemp = temperatures[i];
            greaterFound = false;
            c = 0;
            while (ti < temperatures.length) {
                if (temperatures[ti] > oTemp) {
                    greaterFound = true;
                    break;
                } else {
                    c++;
                }
                ti++;
            }
            result[i] = greaterFound ? c : 0;
        }
        return result;
    }
}
